src/libsystemd/sd-netlink/sd-netlink.c

```diff
@@ -144,7 +144,10 @@ int sd_netlink_open(sd_netlink **ret) {
         return 0;
 }
 
-int sd_netlink_inc_rcvbuf(const sd_netlink *const rtnl, const int size) {
+int sd_netlink_inc_rcvbuf(sd_netlink *rtnl, size_t size) {
```

Comment:
Well, the fact that sd_netlink_inc_rcvbuf() works the way it works really shouldn't be exposed in the API like this. I mean, it might very well be implemented in a "lazy" way, i.e. where we store the value in some internal variable, and only update the setting for the socket whenever we do work again, but if we encode in the public API that the object is "const", then this is not possible. (or if the "lazy" argument doesn't count in your eyes, we might very well want to "shadow" this variable eventually, to size buffers correctly, but we can't if we declared it "const" earlier). 

Also, in this case, conceptually, the call *does* modify the object (specifically: a sub-object owned by it, encapsulated in the socket fd), it doesn't alter the local memory for that, but conceptually I'd claim that in this case it is highly misleading to claim that the object wouldn't be modified...

I documented this general case in the CODING_STYLE update included in #3432 this now,